modules/overrides/interface.go

```diff
@@ -0,0 +1,57 @@
+package overrides
+
+import (
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/grafana/dskit/services"
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/grafana/tempo/pkg/sharedconfig"
+	"github.com/grafana/tempo/pkg/spanfilter/config"
+)
+
+type Service interface {
+	services.Service
+	prometheus.Collector
+	Interface
+
+	WriteStatusRuntimeConfig(w io.Writer, r *http.Request) error
+}
+
+type Interface interface {
```

Comment:
this is a really intense interface which will make implementation/mocking difficult. is there anything we can do to simplify?

maybe:
```
type Interface interface {
  Limit(type, userID) 
}
```
That doesn't seem to work b/c each limit returns a different type and returning empty interface is kind of gross.

Is there a better point of abstraction? maybe something inside the overrides module?

Reply:
Interesting, I think extending/reusing code of the runtimeconfig.Manager to work with other sources would be neat. The polling logic will be needed for every source I think.

Besides fetching the limits we also need logic to merge the overrides of multiple sources and I think that will still end up somewhere in Tempo and not in dskit since dskit isn't aware of our limits. Adding a first layer of abstraction (the `overrides.Interface`) will make it easier to put a component in between that can deal with the merging logic.

Reply:
understood :+1: